materials/variation_with_state_test.py

```diff
@@ -0,0 +1,278 @@
+            ['temperature'], {'temperature': 'kelvin'}, 'multiplier', 'reference',
+            np.arange(4), np.arange(4)**2,
+            ['linear'])
+
+        # Action
+        domain = state_model.get_state_domain()
+
+        # Verification
+        self.assertEqual(len(domain), 1)
+        self.assertTrue('temperature' in domain)
+        self.assertEqual(domain['temperature'][0], 0)
+        self.assertEqual(domain['temperature'][1], 3)
+
+
+    def test_domain_2d(self):
+        """Test get_state_domain on a 2-d lookup table."""
+        # Setup
+        yaml_dict = {
+            'exposure time': {
+                0.0: {
+                    'temperature': np.arange(4),
+                    'values': np.arange(4)**2
+                },
+                0.1: {
+                    'temperature': np.arange(4),
+                    'values': (np.arange(4) + 0.5)**2
+                },
+            }
+        }
+        state_vars = ['exposure time', 'temperature']
+        scales = ['linear', 'linear']
+        #pylint: disable=protected-access
+        interp_points, interp_values = vstate._create_interp_arrays_from_yaml_table(
+            yaml_dict, state_vars, scales)
+
+        state_model = vstate.VariationWithStateTable(
+            state_vars, {'exposure time': 'hour', 'temperature': 'kelvin'}, 'multiplier', 'reference',
+            interp_points, interp_values,
+            scales)
+
+        # Action
+        domain = state_model.get_state_domain()
+
+        # Verification
+        self.assertEqual(len(domain), 2)
+        self.assertTrue('exposure time' in domain)
+        self.assertTrue('temperature' in domain)
+        self.assertEqual(domain['exposure time'][0], 0.0)
+        self.assertEqual(domain['exposure time'][1], 0.1)
+        self.assertEqual(domain['temperature'][0], 0)
+        self.assertEqual(domain['temperature'][1], 3)
+
+    def test_str(self):
+        """Unit test for __str__."""
+        # Setup
+        yaml_dict = {
+            'exposure time': {
+                0.0: {
+                    'temperature': np.arange(4),
+                    'values': np.arange(4)**2
+                },
+                0.1: {
+                    'temperature': np.arange(4),
+                    'values': (np.arange(4) + 0.5)**2
+                },
+            }
+        }
+        state_vars = ['exposure time', 'temperature']
+        scales = ['linear', 'linear']
+        #pylint: disable=protected-access
+        interp_points, interp_values = vstate._create_interp_arrays_from_yaml_table(
+            yaml_dict, state_vars, scales)
+
+        state_model = vstate.VariationWithStateTable(
+            state_vars, {'exposure time': 'hour', 'temperature': 'kelvin'}, 'multiplier', 'reference',
+            interp_points, interp_values,
+            scales)
+
+        # Action
+        string = str(state_model)
+
+        # Verification
+        # TODO automate checking this?
+        print(string)
```

Comment:
Sinful!

Reply:
Fixed in 380275cfe343ffa2b7bf4ce9b3f40cdd31abf183